Formula/snap.rb

```diff
@@ -0,0 +1,45 @@
+class Snap < Formula
+  desc "Tool to work with .snap files"
+  homepage "https://snapcraft.io/"
+  url "https://github.com/snapcore/snapd/releases/download/2.46/snapd_2.46.vendor.tar.xz"
+  sha256 "c4f532018ca9d2a5f87a95909b3674f8e299e97ba5cb5575895bcdd29be23db3"
+  license "GPL-3.0-only"
+  head "https://github.com/snapcore/snapd.git"
+
+  depends_on "go" => :build
+  depends_on "squashfs"
+
+  def install
+    ENV["GOPATH"] = buildpath
+    (buildpath/"src/github.com/snapcore/snapd").install buildpath.children
+
+    cd "src/github.com/snapcore/snapd" do
+      if version.head?
+        system "./get-deps.sh"
```

Comment:
What does this download?

Reply:
this essentially just installs govendor and the deps declared in vendor.json using govendor, you could call govendor directly if one was so inclined

Reply:
Any downloads needs to be resource blocks or using a check-summed system.

Reply:
I'll let @sergiusens sort out how to do that in this recipe, just wanted to clarify what that script does as a snapd developer myself. Note also that govendor does checksum things when it downloads them, the vendor.json contains the hashes: https://github.com/snapcore/snapd/blob/master/vendor/vendor.json

Reply:
@anonymouse64 any reason why you don't switch to Go Modules?

Reply:
We have to support very old Linux distros, we want to switch but can't yet until old distros have backported newer go versions

Reply:
To clarify, this is only required for `--HEAD`, which I can remove from the formula if this becomes a blocker. The "released path"  is using vendored assets.

Reply:
I'd say that's the better solution here.

Reply:
I have removed head and with it the call to get-deps.sh